pykafka/producer.py

```diff
@@ -349,6 +350,7 @@ def mark_as_delivered(message_batch):
                 for partition, presponse in iteritems(partitions):
                     if presponse.err == 0:
                         mark_as_delivered(req.msets[topic][partition].messages)
+                        delivered += len(req.msets[topic][partition].messages)
```

Comment:
I think this counter increment should go inside `mark_as_delivered`, to keep the logic dependent on `req.msets[topic][partition].messages` in one place.